reservation/reservation.go

```diff
@@ -106,6 +106,16 @@ func (manager *Manager) Lock(resource string) (*Reservation, error) {
 	return res, nil
 }
 
+// WaitUntilLock creates a Reservation for `resource`, or waits until it can do so.
+func (manager *Manager) WaitUntilLock(resource string) (*Reservation, error) {
+	res, err := manager.Lock(resource)
+	for err != nil && err.Error() == fmt.Sprintf("Reservation already exists for resource %s", resource) {
```

Comment:
I don't like that attempt at a do/while as much.  I do just wish I had a do/while, but of the two ways I prefer mine and find it a little easier to read.  I will take the `Sprintf` suggestion, though!